llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -960,10 +988,11 @@ bool AIEPreRASchedStrategy::tryCandidate(SchedCandidate &Cand,
       if (!PC.isValid())
         return false;
       unsigned CurrPressure = BotRPT.getRegSetPressureAtPos()[PC.getPSet()];
-      unsigned Threshold =
-          TRI->getRegPressureSetLimit(*CurMBB->getParent(), PC.getPSet());
-      return Threshold <= NumCriticalFreeRegs ||
-             CurrPressure >= Threshold - NumCriticalFreeRegs;
+      unsigned Threshold = PSetThresholds[PC.getPSet()];
+      unsigned NumCriticalFreeUnits =
+          NumCriticalFreeRegs * std::abs(PC.getUnitInc());
```

Comment:
Curious: under what circumstances would getUnitInc be negative?

Reply:
It will be negative if the instruction reduces pressure

Reply:
Ah, I missed that this is per instruction. Thx.